### ISSUE TITLE
Core Dump

### ISSUE THREAD

**Issue by internalfx** [rethinkdb/rethinkdb#3393]
Not sure what to make of it exactly, I do use change feeds....

Server was under high load when it happened.

```
Version: rethinkdb 1.15.2~0trusty (GCC 4.8.2)
error: Error in src/concurrency/auto_drainer.cc at line 24:
error: Guarantee failed: [!parent->draining.is_pulsed()] New processes should not acquire a draining `auto_drainer_t`.
error: Backtrace:
addr2line: 'rethinkdb': No such file
addr2line: 'rethinkdb': No such file
error: Wed Dec  3 13:54:07 2014

       1: backtrace_t::backtrace_t() at 0xb927b0 (rethinkdb)
       2: format_backtrace(bool) at 0xb92b43 (rethinkdb)
       3: report_fatal_error(char const*, int, char const*, ...) at 0xd7ed35 (rethinkdb)
       4: auto_drainer_t::lock_t::lock_t(auto_drainer_t*) at 0xb9b1e8 (rethinkdb)
       5: ql::changefeed::feed_t::each_sub_in_vec(std::vector<std::set<ql::changefeed::subscription_t*, std::less<ql::changefeed::subscrip
tion_t*>, std::allocator<ql::changefeed::subscription_t*> >, std::allocator<std::set<ql::changefeed::subscription_t*, std::less<ql::change
feed::subscription_t*>, std::allocator<ql::changefeed::subscription_t*> > > > const&, rwlock_in_line_t*, std::function<void (ql::changefee
d::subscription_t*)> const&) at 0xa319b5 (rethinkdb)
       6: ql::changefeed::feed_t::each_table_sub(std::function<void (ql::changefeed::subscription_t*)> const&) at 0xa31b35 (rethinkdb)
       7: ql::changefeed::msg_visitor_t::operator()(ql::changefeed::msg_t::change_t const&) const at 0xa3bc53 (rethinkdb)
       8: ql::changefeed::feed_t::mailbox_cb(ql::changefeed::stamped_msg_t) at 0xa32691 (rethinkdb)
       9: std::_Function_handler<void (ql::changefeed::stamped_msg_t), std::_Bind<std::_Mem_fn<void (ql::changefeed::feed_t::*)(ql::change
feed::stamped_msg_t)> (ql::changefeed::feed_t*, std::_Placeholder<1>)> >::_M_invoke(std::_Any_data const&, ql::changefeed::stamped_msg_t) 
at 0xa3923d (rethinkdb)
       10: mailbox_t<void (ql::changefeed::stamped_msg_t)>::read_impl_t::read(cluster_version_t, read_stream_t*) at 0xa3b6a5 (rethinkdb)
       11: mailbox_manager_t::mailbox_read_coroutine(connectivity_cluster_t::connection_t*, auto_drainer_t::lock_t, threadnum_t, unsigned 
long, std::vector<char, std::allocator<char> >*, long, mailbox_manager_t::force_yield_t) at 0xaefadb (rethinkdb)
       12: rethinkdb() [0xaefbd2] at 0xaefbd2 ()
       13: coro_t::run() at 0xad11c8 (rethinkdb)
error: Exiting.
Trace/breakpoint trap (core dumped)
```

**Comment by AtnNn** [comment]
@internalfx The fix has been release in RethinkDB 1.15.3

**Comment by internalfx** [comment]
@danielmewes @mlucy @AtnNn You guys never cease to impress.
